src/node.cc

```diff
@@ -4202,17 +4201,19 @@ inline void PlatformInit() {
 
     s.flags = GetFileDescriptorFlags(fd);
     CHECK_NE(s.flags, -1);
+    
+    if (isatty(fd)) {
+      do {
+        err = tcgetattr(fd, &s.termios);
+      } while (err == -1 && errno == EINTR);
 
-    do {
-      err = tcgetattr(fd, &s.termios);
-    } while (err == -1 && errno == EINTR);
-
-    if (err == 0) {
-      s.isatty = true;
-    } else {
-      // tcgetattr() is not supposed to return ENODEV or EOPNOTSUPP
-      // but it does so anyway on MacOS.
-      CHECK(errno == ENODEV || errno == ENOTTY || errno == EOPNOTSUPP);
+      if (err == 0) {
+        s.isatty = true;
+      } else {
+        // tcgetattr() is not supposed to return ENODEV or EOPNOTSUPP
+        // but it does so anyway on MacOS.
+        CHECK(errno == ENODEV || errno == ENOTTY || errno == EOPNOTSUPP);
+      }
```

Comment:
This can probably be simplified to just this now:
```c
s.isatty = true;
do {
  err = tcgetattr(fd, &s.termios);
} while (err == -1 && errno == EINTR);
CHECK_EQ(err, 0);
```